relay-filter/src/health_check.rs

```diff
@@ -0,0 +1,162 @@
+//! Implements event filtering based on whether the endpoint called is a healthcheck endpoint
+//!
+//! If this filter is enabled messages to healthcheck endpoints will be filtered out
+
+use once_cell::sync::Lazy;
+use regex::Regex;
+use relay_general::protocol::Event;
+
+use crate::{FilterConfig, FilterStatKey};
+
+static HEALTH_CHECK_ENDPOINTS: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(
+        r#"(?ix)
+        healthcheck|
+        healthy|
+        live|
+        ready|
+        heartbeat|
+        /health$|
+        /healthz$
+        "#,
+    )
+    .expect("Invalid healthcheck filter Regex")
+});
+
+fn matches(_event: &Event) -> bool {
+    if let Some(transaction) = _event.transaction.value() {
+        HEALTH_CHECK_ENDPOINTS.is_match(transaction)
+    } else {
+        false
+    }
```

Comment:
I think we can simplify this a bit:
```suggestion
    _event.transaction.value().map_or(false, |transaction| {
        HEALTH_CHECK_ENDPOINTS.is_match(transaction)
    })
```